pkg/image/api/types.go

```diff
@@ -60,6 +84,69 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureState holds a result of signature's server-side check.
+type SignatureState string
+
+// SignatureForImage describes relation of the signature to the image containing it.
+type SignatureForImage string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side signature check. IOW a time of the last update of SignatureState.
+	Checked *unversioned.Time
+	// Says whether the signature matches image object containing it.
+	ForImage SignatureForImage
+	// Result of the last server-side signature check performed at Checked. Filled by server.
+	State SignatureState
```

Comment:
As you noted in the gist, you could collapse this, `ForImage`, and `Checked` into two different `Conditions` like the `[]PodCondition`